lazy_dataset/core.py

```diff
@@ -2720,27 +2717,34 @@ def _get_memory_size(keep_mem_free):
             import humanfriendly
             return humanfriendly.parse_size(keep_mem_free, binary=True)
 
+    def check(self):
+        if self._keep_mem_free is None:
+            return True
+
+        import psutil
+        if psutil.virtual_memory().available <= self._keep_mem_free:
+            # Return without writing to cache if there is not enough
+            # free memory
+            import warnings
+            warnings.warn(
+                'Max capacity of the in-memory cache is reached. '
+                'The remaining data will not be cached.'
+            )
+            return False
```

Comment:
Can you introduce a short circuit? One warning is enough and after one warning you can assume that the memory is full.

Reply:
Done